src/main/gui/main-window.ts

```diff
@@ -24,13 +24,41 @@ import { BackendProcess } from '../backend/process';
 import { setupBackend } from '../backend/setup';
 import { isArmMac, isMac } from '../env';
 import { addBrowserWindow, addFile, addFiles, removeFile, removeFiles } from '../fileWatcher';
-import { getRootDir } from '../platform';
+import { getIsPortableSync, getRootDir, installDir } from '../platform';
 import { BrowserWindowWithSafeIpc, ipcMain } from '../safeIpc';
-import { SaveFile, openSaveFile } from '../SaveFile';
+import { SaveData, SaveFile, openSaveFile } from '../SaveFile';
 import { writeSettings } from '../setting-storage';
+import { getAllFiles } from '../util';
 import { MenuData, setMainMenu } from './menu';
 
 const version = app.getVersion() as Version;
+const documentsDir = app.getPath('documents');
+
+const askSaveLocation = async (
+    mainWindow: BrowserWindowWithSafeIpc,
+    saveData: SaveData,
+    defaultPath: string | undefined
+): Promise<FileSaveResult> => {
+    const { canceled, filePath } = await dialog.showSaveDialog(mainWindow, {
+        title: 'Save Chain File',
+        filters: [{ name: 'Chain File', extensions: ['chn'] }],
+        defaultPath: defaultPath ?? documentsDir,
+    });
+    if (!canceled && filePath) {
+        if (filePath.startsWith(installDir)) {
+            dialog.showMessageBoxSync({
```

Comment:
This shouldn't be sync, it'll block the entire app.

And should also make this message a child(?) of the main window. Otherwise, users can just keep the message open and interact with the main window.